src/HealthChecks.Nats/NatsHealthCheck.cs

```diff
@@ -1,111 +1,36 @@
-using System.Text;
 using Microsoft.Extensions.Diagnostics.HealthChecks;
-using NATS.Client;
+using NATS.Client.Core;
 
 namespace HealthChecks.Nats;
 
 /// <summary>
 /// Health check for Nats Server.
 /// </summary>
-/// <remarks>
-/// Relies on a static <see cref="ConnectionFactory"/> which provides factory methods to create
-/// connections to NATS Servers, and a <see cref="IConnection"/> object connected to the NATS server.
-/// </remarks>
-public sealed class NatsHealthCheck : IHealthCheck, IDisposable
+public sealed class NatsHealthCheck(INatsConnection connection) : IHealthCheck
 {
-    private static readonly ConnectionFactory _connectionFactory = new();
-
-    private readonly NatsOptions _options;
-
-    private IConnection? _connection;
-
-    public NatsHealthCheck(NatsOptions natsOptions)
+    public async Task<HealthCheckResult> CheckHealthAsync(HealthCheckContext context, CancellationToken cancellationToken = default)
     {
-        _options = Guard.ThrowIfNull(natsOptions);
+        var result = connection.ConnectionState switch
```

Comment:
How about just always calling `ConnectAsync()`?

It will achieve the same effect:
- It will just return a finished task for `Open` connection
- try to re-open the closed connections which should help us avoid getting into issues like https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/issues/1544

But also wait for the reconnection to finish for `Connecting` and `Reconnecting`. In case the enum is ever extended with a new value, `ConnectAsync()` will most likely handle it for us.

https://github.com/nats-io/nats.net/blob/3f919ddcd55ea0f765b4ce9b4137e2a58c589683/src/NATS.Client.Core/NatsConnection.cs#L155-L183